transports/email/src/main/java/org/mule/transport/email/RetrieveMessageReceiver.java

```diff
@@ -166,11 +162,50 @@ protected void doStop() throws MuleException
     @Override
     protected void doStart() throws MuleException
     {
-        super.doStart();
-        synchronized (folderLock)
+        try
         {
-            folder.addMessageCountListener(this);
+            getConnector().getRetryPolicyTemplate().execute(new RetryCallback()
```

Comment:
Add a comment explaining why this deferring is needed.

Reply:
Comment added